django/urls/base.py

```diff
@@ -179,3 +184,89 @@ def translate_url(url, lang_code):
+        return URLPattern(pattern, view, kwargs, name)
+    else:
+        raise TypeError('view must be a callable or a list/tuple in the case of include().')
+
+
+def path(regex, view, kwargs=None, name=None):
+    if isinstance(view, (list, tuple)):
+        # For include(...) processing.
+        pattern = RoutePattern(regex, is_endpoint=False)
+        urlconf_module, app_name, namespace = view
+        return URLResolver(
+            pattern,
+            urlconf_module,
+            kwargs,
+            app_name=app_name,
+            namespace=namespace,
+        )
+    elif callable(view):
+        pattern = RoutePattern(regex, name=name, is_endpoint=True)
+        return URLPattern(pattern, view, kwargs, name)
+    else:
+        raise TypeError('view must be a callable or a list/tuple in the case of include().')
+
+
+def include(arg, namespace=None):
+    app_name = None
+    if isinstance(arg, tuple):
+        # callable returning a namespace hint
+        try:
+            urlconf_module, app_name = arg
+        except ValueError:
+            if namespace:
+                raise ImproperlyConfigured(
+                    'Cannot override the namespace for a dynamic module that '
+                    'provides a namespace.'
+                )
+            raise ImproperlyConfigured(
+                'Passing a %d-tuple to django.conf.urls.include() is not supported. '
+                'Pass a 2-tuple containing the list of patterns and app_name, '
+                'and provide the namespace argument to include() instead.' % len(arg)
+            )
+    else:
+        # No namespace hint - use manually provided namespace
+        urlconf_module = arg
+
+    if isinstance(urlconf_module, str):
+        urlconf_module = import_module(urlconf_module)
+    patterns = getattr(urlconf_module, 'urlpatterns', urlconf_module)
+    app_name = getattr(urlconf_module, 'app_name', app_name)
+    if namespace and not app_name:
+        raise ImproperlyConfigured(
+            'Specifying a namespace in django.conf.urls.include() without '
+            'providing an app_name is not supported. Set the app_name attribute '
+            'in the included module, or pass a 2-tuple containing the list of '
+            'patterns and app_name instead.',
+        )
+
+    namespace = namespace or app_name
+
+    # Make sure we can iterate through the patterns (without this, some
+    # testcases will break).
+    if isinstance(patterns, (list, tuple)):
+        for url_pattern in patterns:
+            # Test if the LocaleRegexURLResolver is used within the include;
+            # this should throw an error since this is not allowed!
+            if isinstance(url_pattern.pattern, LocalePrefixPattern):
+                raise ImproperlyConfigured(
+                    'Using i18n_patterns in an included URLconf is not allowed.')
+
+    return (urlconf_module, app_name, namespace)
```

Comment:
I didn't review this in detail because it appears to be largely moved from `django.conf.urls`.

I skimmed it without spotting anything shocking.